### ISSUE TITLE
[BUG] Fails to load in Unraid 6.11.0-RC1

### ISSUE THREAD

**Issue by NickM-27** [dmacias72/unRAID-NerdPack#69]
Updated to Unraid 6.11.0-RC1 and it never seems to load, gives this error in the logs:

`root: Warning: preg_grep() expects parameter 2 to be array, string given in /usr/local/emhttp/plugins/NerdPack/scripts/packagemanager on line 79`

**Comment by alez007** [comment]
it's because no package folder has been added yet for unraid version 6.11. I needed this to work ASAP so I hacked into it by modifying `/usr/local/emhttp/plugins/NerdPack/include/NerdPackHelpers.php` and specifically adding version 6.10 on line 3

<img width="898" alt="Screenshot 2022-07-30 at 21 16 18" src="https://user-images.githubusercontent.com/459478/181994883-96b1a434-f5e3-4340-92ed-43fa5a6c7290.png">

it's a hack I know 🤷  but I'm sure this will get fixed soon

**Comment by weirdcrap** [comment]
> it's because no package folder has been added yet for unraid version 6.11. I needed this to work ASAP so I hacked into it by modifying `/usr/local/emhttp/plugins/NerdPack/include/NerdPackHelpers.php` and specifically adding version 6.10 on line 3
> 
> <img alt="Screenshot 2022-07-30 at 21 16 18" width="898" src="https://user-images.githubusercontent.com/459478/181994883-96b1a434-f5e3-4340-92ed-43fa5a6c7290.png">
> 
> it's a hack I know shrug but I'm sure this will get fixed soon
Does this persist between reboots? I'd like to be on the RC as it supposedly addresses an issue I've been having but I also heavily rely on a few tools from NerdPack.

Looking forward to an official fix.

**Comment by weirdcrap** [comment]
> it's because no package folder has been added yet for unraid version 6.11. I needed this to work ASAP so I hacked into it by modifying `/usr/local/emhttp/plugins/NerdPack/include/NerdPackHelpers.php` and specifically adding version 6.10 on line 3
> 
> <img alt="Screenshot 2022-07-30 at 21 16 18" width="898" src="https://user-images.githubusercontent.com/459478/181994883-96b1a434-f5e3-4340-92ed-43fa5a6c7290.png">
> 
> it's a hack I know shrug but I'm sure this will get fixed soon
Does this persist between reboots? I'd like to be on the RC as it supposedly addresses an issue I've been having but I also heavily rely on a few tools from NerdPack.

Looking forward to an official fix.

**Comment by alez007** [comment]
> > it's because no package folder has been added yet for unraid version 6.11. I needed this to work ASAP so I hacked into it by modifying `/usr/local/emhttp/plugins/NerdPack/include/NerdPackHelpers.php` and specifically adding version 6.10 on line 3
> > <img alt="Screenshot 2022-07-30 at 21 16 18" width="898" src="https://user-images.githubusercontent.com/459478/181994883-96b1a434-f5e3-4340-92ed-43fa5a6c7290.png">
> > it's a hack I know shrug but I'm sure this will get fixed soon
> 
> Does this persist between reboots? I'd like to be on the RC as it supposedly addresses an issue I've been having but I also heavily rely on a few tools from NerdPack.
> 
> Looking forward to an official fix.

I haven't tried it but I doubt it persists between reboots. It's just a hack I'm using until a proper fix is in place.

**Comment by weirdcrap** [comment]
So the discussion on the forum is that dmacias is AFK and this will most likely be marked incompatible in CA with 6.11 stable. /boot/extra appears to be the way to go for installing slackware packages you want.

Unraid is currently on slackware 15 (this can be checked with cat /etc/slackware-version).

**Comment by txwireless** [comment]
Since it looks like this may be a longer term fix, I decided to make the update of the package file more automated in case it doesn't persist the reboot.  Added the following to my go file.

# Fix NerdPack and DevPack for 6.11
if [ -d /usr/local/emhttp/plugins/NerdPack ]; then
  sed -i 's:^\$os_version.*$:\$os_version = 6.10;:' /usr/local/emhttp/plugins/NerdPack/include/NerdPackHelpers.php
fi
if [ -d /usr/local/emhttp/plugins/DevPack ]; then
  sed -i 's:^\$os_version.*$:\$os_version = 6.10;:' /usr/local/emhttp/plugins/DevPack/include/DevPackHelpers.php;
fi

**Comment by alez007** [comment]
> Since it looks like this may be a longer term fix, I decided to make the update of the package file more automated in case it doesn't persist the reboot. Added the following to my go file.
> 
> ![DevPack and NerdPack Fix](https://user-images.githubusercontent.com/15952375/185229056-cd77355b-b673-4104-b14f-188d76c0af35.jpg)

I've been less elegant than that: 

```
cp /boot/config/plugins/NerdPack/packages/6.10/packages.json /boot/config/plugins/NerdPack/packages/6.11/packages.json
```

**Comment by thorrrr** [comment]
HI Guys
I am getting a warning re this plugin
The author (or moderators of Community Applications) of the plugin template (https://raw.githubusercontent.com/dmacias72/unRAID-NerdPack/master/plugin/NerdPack.plg) has specified that this plugin is incompatible with your version of Unraid (6.11.0). You should uninstall the plugin here

Do i need to remove it? I thought reading above it was type of fixed for now but now not sure!

**Comment by thecode** [comment]
> HI Guys I am getting a warning re this plugin The author (or moderators of Community Applications) of the plugin template (https://raw.githubusercontent.com/dmacias72/unRAID-NerdPack/master/plugin/NerdPack.plg) has specified that this plugin is incompatible with your version of Unraid (6.11.0). You should uninstall the plugin here
> 
> Do i need to remove it? I thought reading above it was type of fixed for now but now not sure!

It is already disabled once you upgraded to 6.11, you should remove it. Keep watching this issue, if a new version released you can reinstall it

**Comment by alez007** [comment]
yeah, the fixes here are more or less hacks, if you can afford waiting for a proper update, then better to do so

**Comment by alez007** [comment]
@thorrrr apparently the owner of this pack is MIA so a new plugin was launched for unraid: https://forums.unraid.net/topic/129200-plug-in-nerdtools/
